lib/events/kafka_listener.rb

```diff
@@ -36,9 +36,9 @@ def raw_process(event)
 
       # Kafka message from Ingress has no headers. We need to prepare the headers from its payload.
       if insights_headers.empty?
-        if event.payload.class == Hash && event.payload.has_key?("params") && event.payload["params"].has_key?("external_tenant")
+        if event.payload.instance_of?(Hash) && event.payload.key?("params") && event.payload["params"].key?("external_tenant")
           # check availability from source doesn't have headers
-          insights_headers['x-rh-insights-request-id'] = "unknown"
+          insights_headers['x-rh-insights-request-id'] = "scheduled_task_#{Time.current.to_s.tr(' ', '_')}"
```

Comment:
@hsong-rh NoMethodError (undefined method `current' for Time:Class) is that a Rails thing.

When I changed it to Time.now I the string looks like
```scheduled_task_2021-02-22_11:54:43_-0500```

Reply:
@hsong-rh  Should we set the date/time to be UTC so we don't have the TimeZone to worry about?